templates/customizations/Dockerfile.j2

```diff
@@ -5,4 +5,10 @@ RUN apt -y update
 RUN apt -y install {{ nextcloud_container_image_customizations_php_imageick_installation_package }}
 {% endif %}
 
+{% if nextcloud_container_image_customizations_smb_enabled %}                                                                          
+RUN apt -y update                                                                                                                        
+RUN apt -y install procps smbclient                                                                                                      
+RUN rm -rf /var/lib/apt/lists/*                                                                                                          
```

Comment:
Deleting files from new container building steps has no effect on space savings. The previous layer contains the files, so the fact that the new layer marks them as deleted does not help (unless the image layers are squished).

To effectively "use apt and delete its cache", you need to do everything in a single step (a single `RUN` statement). Example:

```dockerfile
RUN apt -y update && apt -y install procps smbclient && rm -rf /var/lib/apt/lists/*
```

-------

To prevent multiple `apt -y update` calls and having to do the same things over again, I suggest that the `Dockerfile.j2` look like this:

```jinja2
FROM {{ nextcloud_container_image }}

{% set packages_to_install = [] %}

{% if nextcloud_container_image_customizations_php_imageick_installation_enabled %}
{% set packages_to_install = packages_to_install + [nextcloud_container_image_customizations_php_imageick_installation_package] %}
{% endif %}

{% if nextcloud_container_image_customizations_samba_enabled %}
{% set packages_to_install = packages_to_install + ['procps', 'smbclient'] %}
{% endif %}

{% if packages_to_install | length > 0 %}                                                                          
RUN apt -y update && apt -y install {{ packages_to_install | join(' ') }} && rm -rf /var/lib/apt/lists/*                                                                                                          
{% endif %}

{{ nextcloud_container_image_customizations_dockerfile_body_custom }}
```

Note: the above is untested.

Reply:
this totally makes sense! still learning docker, but the comment about each RUN causing a new image layer, so the need to put the rm in the same line, really helps me understand the approach. 

ok, i've tested all the above changes with my local setup and it works, so creating a revised PR now.